keps/sig-api-machinery/0015-dry-run.md

```diff
@@ -150,3 +151,52 @@ A dry-run request should behave as close as possible to a regular
 request. Attempting to dry-run create an existing object will result in an
 `AlreadyExists` error to be returned. Similarly, if a dry-run update is
 performed on a non-existing object, a `NotFound` error will be returned.
+
+## kubectl
+
+For the `kubectl` client integration with server-side dry-run, we will pass
+the `dryRun` query parameter by reading the user's intent from a flag.
+
+For beta, we use `--server-dry-run` for `kubectl apply` to exercise
+server-side apply. This flag will be deprecated next release, then removed after 1 release.
+
+For GA, we'll use the existing `--dry-run` flag available on `kubectl apply`.
+
+Currently, the `--dry-run` flag is a boolean for subcommands including
+`kubectl apply` and `kubectl create`.
+
+We'll extend the flag to accept strings for new options
+for selecting client-side and server-side behavior: `client`, `server`, and
+`none`.
+
+For backwards compatibility, we'll continue to default the value for `--dry-run`
+to `--dry-run=client`, which is equivalent to the existing behavior for
+`--dry-run=true`.
+
+The boolean values for `--dry-run` will be deprecated next release, then removed in 2 releases.
+
+The `--dry-run` flag will be parsed as described below.
+
+```golang
+func ParseDryRun(dryRunFlag string) (DryRunEnum, error) {
+  b, err := strconv.ParseBool(dryRunFlag)
+  if err != nil { // The flag is not a boolean
+    switch dryRunFlag {
```

Comment:
I expected to see handling for true and false, which are I would expect to be allowed during the transition.  Rare, but allowed.

Reply:
> I expected to see handling for true and false, which are I would expect to be allowed during the transition. Rare, but allowed.

ah, didn't read the line above.  retracted.